src/pages/[guild]/collect/[chain]/[address].tsx

```diff
@@ -182,12 +193,32 @@ const Page = ({
                     >
                       {name}
                     </Heading>
+
                     {isMobile && <RequirementsCard role={role} />}
 
                     <Box ref={nftDescriptionRef} lineHeight={1.75}>
-                      <RichTextDescription
-                        text={guildPlatform?.platformGuildData?.description}
-                      />
+                      <HStack alignItems="start" justifyContent="flex-end">
+                        <RichTextDescription
+                          text={guildPlatform?.platformGuildData?.description}
+                        />
+                        <Spacer m="0" />
+                        {isAdmin && (
+                          <>
+                            <IconButton
+                              icon={<Icon as={PencilSimple} />}
+                              size="sm"
+                              rounded="full"
+                              aria-label="Edit description"
+                              onClick={onOpen}
+                            />
+                            <EditNFTDescriptionModal
```

Comment:
Since this component is only loaded for admins, I'd suggest you to load it dynamically only on client-side with `next/dynamic`, you can find examples in our codebase, but let me know if you have any questions!
You could event create an `EditNFTDescriptionButton` component with the `useDisclosure` hook (in that case you won't need to rename the isOpen, onOpen, onClose props!) + the `IconButton` &  `EditNFTDescriptionModal`